### PR TITLE
[FIX] ReliefF: Support for missing target values

### LINKED ISSUE #1046: Rank widgets crashes when some target values are missing
I tried this with regression data; classification probably has the same problem.

### PR DESCRIPTION
**Pull request by matevzkren**
Fixed rank widget crashing on missing target class values. Affects both
ReliefF and RReliefF. Fixes #1046.

### CODE REVIEW

File: Orange/preprocess/_relieff.pyx [hunk 1]
```diff
@@ -365,6 +365,9 @@ cdef tuple prepare(X, y, is_discrete, Contingencies &contingencies):
         X[:, is_continuous] -= row_min[is_continuous]
         X[:, is_continuous] /= row_ptp[is_continuous]
     y = np.array(y, dtype=np.float64)
+    isan = np.logical_not(np.isnan(y))
```
Comment (kernc) [inline Orange/preprocess/_relieff.pyx]: `isan` is nondescript. Rename to `is_defined` (likened to the surrounding code), `nonnan`, or similar.

**Comment by kernc** [comment]
Great, please [squash the commits](https://github.com/biolab/orange3/blob/master/CONTRIBUTING.md#pull-requests) into one and use the commit message of the latter one.

**Comment by kernc** [comment]
Umm, could you also supply a test case for one of the ReliefFs that covers this edge case?

**Comment by kernc** [comment]
@lanzagar I think this is ready for merge.

**Comment by lanzagar** [comment]
Looks good to me!
